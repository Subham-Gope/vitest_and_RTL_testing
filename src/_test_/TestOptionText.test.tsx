// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import TestOptionText from "../components/TestOptionText";

// describe("Testing the TestOptionText.tsx file", () => {
//   test("selects multiple values", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     // setting up the debugger
//     screen.debug();

//     // setting up snapshot matching
//     const { container } = render(<TestOptionText />);
//     expect(container).toMatchSnapshot();

//     const select = screen.getByRole("listbox", {
//       name: /select a option/i,
//     }) as HTMLSelectElement;
//     expect(select).toBeInTheDocument();

//     // Initiating user actions
//     await user.selectOptions(select, ["Mango", "Papaya"]);

//     const selectedOptions = Array.from(
//       select.selectedOptions,
//       (option) => option.value
//     );

//     expect(selectedOptions).toEqual(["mango", "papaya"]);

//     await user.selectOptions(select, ["Bannana", "Pineapple"]);

//     const selectedOptions2 = Array.from(
//       select.selectedOptions,
//       (option) => option.value
//     );
//     expect(selectedOptions2).toEqual([
//       "mango",
//       "bannana",
//       "pineapple",
//       "papaya",
//     ]);

//     await user.deselectOptions(select, ["pineapple", "papaya"]);

//     const selectedOptions3 = Array.from(
//       select.selectedOptions,
//       (option) => option.value
//     );
//     expect(selectedOptions3).toEqual(["mango", "bannana"]);
//   });
// });
