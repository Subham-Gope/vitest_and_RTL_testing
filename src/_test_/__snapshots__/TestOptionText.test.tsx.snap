// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the TestOptionText.tsx file > selects multiple values 1`] = `
<div>
  <div>
    <label
      for="optionText"
    >
      Select a Option:
    </label>
    <select
      id="optionText"
      multiple=""
    >
      <option
        value="mango"
      >
        Mango
      </option>
      <option
        value="bannana"
      >
        Bannana
      </option>
      <option
        value="pineapple"
      >
        Pineapple
      </option>
      <option
        value="papaya"
      >
        Papaya
      </option>
    </select>
  </div>
  <p>
    Selected Values: 
  </p>
</div>
`;
